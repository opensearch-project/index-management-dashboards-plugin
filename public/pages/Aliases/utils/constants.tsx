/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { SortDirection } from "../../../utils/constants";
import { IAlias } from "../interface";

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: "0",
  size: "20",
  search: "",
  sortField: "alias" as keyof IAlias,
  sortDirection: SortDirection.DESC,
  status: "",
  dataSourceId: "",
};
