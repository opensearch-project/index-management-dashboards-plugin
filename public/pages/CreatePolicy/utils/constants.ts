/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "A simple default policy that changes the replica count between hot and cold states.",
      default_state: "example_hot_state",
      states: [
        {
          name: "example_hot_state",
          actions: [{ replica_count: { number_of_replicas: 5 } }],
          transitions: [{ state_name: "example_cold_state", conditions: { min_index_age: "30d" } }],
        },
        {
          name: "example_cold_state",
          actions: [{ replica_count: { number_of_replicas: 2 } }],
          transitions: [],
        },
      ],
      ism_template: {
        index_patterns: ["example-index-*"],
      },
    },
  },
  null,
  4
);
