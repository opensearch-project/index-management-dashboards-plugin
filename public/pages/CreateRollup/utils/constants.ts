/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const EMPTY_ROLLUP = JSON.stringify({
  rollup: {
    continuous: false,
    description: "",
    dimensions: [
      {
        date_histogram: {
          source_field: "",
          fixed_interval: "1h",
          timezone: "UTC",
        },
      },
    ],
    enabled: true,
    metrics: [],
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type === undefined ? "-" : type),
  },
];

export const WHERE_BOOLEAN_FILTERS = [
  { text: "Select value", value: "" },
  { text: "True", value: true },
  { text: "False", value: false },
];

export const OPERATORS_MAP = {
  IS: "is",
  IS_NOT: "is_not",
  IS_NULL: "is_null",
  IS_NOT_NULL: "is_not_null",
  IS_GREATER: "is_greater",
  IS_GREATER_EQUAL: "is_greater_equal",
  IS_LESS: "is_less",
  IS_LESS_EQUAL: "is_less_equal",
  STARTS_WITH: "starts_with",
  ENDS_WITH: "ends_with",
  CONTAINS: "contains",
  NOT_CONTAINS: "does_not_contains",
  IN_RANGE: "in_range",
  NOT_IN_RANGE: "not_in_range",
};

export const COMPARISON_OPERATORS = [
  { text: "is", value: OPERATORS_MAP.IS, dataTypes: ["number", "text", "keyword", "boolean"] },
  {
    text: "is not",
    value: OPERATORS_MAP.IS_NOT,
    dataTypes: ["number", "text", "keyword", "boolean"],
  },
  {
    text: "is null",
    value: OPERATORS_MAP.IS_NULL,
    dataTypes: ["number", "text", "keyword", "boolean"],
  },
  {
    text: "is not null",
    value: OPERATORS_MAP.IS_NOT_NULL,
    dataTypes: ["number", "text", "keyword"],
  },
  { text: "is greater than", value: OPERATORS_MAP.IS_GREATER, dataTypes: ["number"] },
  { text: "is greater than equal", value: OPERATORS_MAP.IS_GREATER_EQUAL, dataTypes: ["number"] },
  { text: "is less than", value: OPERATORS_MAP.IS_LESS, dataTypes: ["number"] },
  { text: "is less than equal", value: OPERATORS_MAP.IS_LESS_EQUAL, dataTypes: ["number"] },
  { text: "is in range", value: OPERATORS_MAP.IN_RANGE, dataTypes: ["number"] },
  { text: "is not in range", value: OPERATORS_MAP.NOT_IN_RANGE, dataTypes: ["number"] },
  { text: "starts with", value: OPERATORS_MAP.STARTS_WITH, dataTypes: ["text", "keyword"] },
  { text: "ends with", value: OPERATORS_MAP.ENDS_WITH, dataTypes: ["text", "keyword"] },
  { text: "contains", value: OPERATORS_MAP.CONTAINS, dataTypes: ["text", "keyword"] },
  { text: "does not contains", value: OPERATORS_MAP.NOT_CONTAINS, dataTypes: ["text"] },
];
