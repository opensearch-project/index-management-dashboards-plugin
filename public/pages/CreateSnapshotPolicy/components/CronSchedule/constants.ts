/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import moment from "moment-timezone";

export const CRON_SCHEDULE_FREQUENCY_TYPE = [
  { value: "hourly", text: "Hourly" },
  { value: "daily", text: "Daily" },
  { value: "weekly", text: "Weekly" },
  { value: "monthly", text: "Monthly" },
  { value: "custom", text: "Custom (Cron expression)" },
];

export const TIMEZONES = moment.tz.names().map((tz) => ({ label: tz, text: tz }));

export const WEEK_DAYS = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
