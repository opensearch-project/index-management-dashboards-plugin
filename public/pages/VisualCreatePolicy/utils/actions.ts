/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { htmlIdGenerator } from "@elastic/eui";
import { RolloverAction, UIAction } from "../../../../models/interfaces";

const makeId = htmlIdGenerator();

export enum ActionType {
  Allocation = "allocation",
  Close = "close",
  Delete = "delete",
  ForceMerge = "force_merge",
  IndexPriority = "index_priority",
  Notification = "notification",
  Open = "open",
  ReadOnly = "read_only",
  ReadWrite = "read_write",
  ReplicaCount = "replica_count",
  Rollover = "rollover",
  Rollup = "rollup",
  Snapshot = "snapshot",
}

export class RolloverUIAction implements UIAction<RolloverAction> {
  id: string;
  action: RolloverAction;
  type = ActionType.Rollover;

  constructor(action: RolloverAction, id: string = makeId()) {
    this.action = action;
    this.id = id;
  }

  content = () => `Rollover`;

  clone = (action: RolloverAction) => new RolloverUIAction(action, this.id);

  render = () => null;

  toAction = () => this.action;
}
