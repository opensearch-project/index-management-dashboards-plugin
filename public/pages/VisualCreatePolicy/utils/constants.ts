/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { AliasAction, AllocationAction } from "../../../../models/interfaces";

export enum ActionType {
  Alias = "alias",
  Allocation = "allocation",
  Close = "close",
  Delete = "delete",
  ForceMerge = "force_merge",
  IndexPriority = "index_priority",
  Notification = "notification",
  Open = "open",
  ReadOnly = "read_only",
  ReadWrite = "read_write",
  ReplicaCount = "replica_count",
  Rollover = "rollover",
  Rollup = "rollup",
  Shrink = "shrink",
  Snapshot = "snapshot",
}

export const DEFAULT_POLICY = {
  description: "A simple default policy that changes the replica count between hot and cold states.",
  default_state: "hot",
  states: [
    {
      name: "hot",
      actions: [{ replica_count: { number_of_replicas: 5 } }],
      transitions: [
        {
          state_name: "cold",
          conditions: { min_index_age: "30d" },
        },
      ],
    },
    {
      name: "cold",
      actions: [{ replica_count: { number_of_replicas: 2 } }],
      transitions: [],
    },
  ],
  ism_template: [],
};

export const EMPTY_DEFAULT_POLICY = {
  description: "A sample description of the policy",
  default_state: "",
  states: [],
  ism_template: [],
};

export const DEFAULT_LEGACY_ERROR_NOTIFICATION = {
  destination: {
    slack: {
      url: "<url>",
    },
  },
  message_template: {
    source: "The index {{ctx.index}} failed during policy execution.",
  },
};

export const DEFAULT_ALIAS: AliasAction = {
  alias: {
    actions: [],
  },
};

export const DEFAULT_ALLOCATION: AllocationAction = {
  allocation: {
    require: {},
    include: {},
    exclude: {},
    wait_for: false,
  },
};

export const DEFAULT_CLOSE = {
  close: {},
};
export const DEFAULT_DELETE = {
  delete: {},
};
export const DEFAULT_FORCE_MERGE = {
  force_merge: {
    max_num_segments: 1,
  },
};
export const DEFAULT_INDEX_PRIORITY = {
  index_priority: {
    priority: 1,
  },
};
export const DEFAULT_NOTIFICATION = {
  notification: {
    channel: {
      id: "",
    },
    message_template: {
      source: "",
    },
  },
};
export const DEFAULT_OPEN = {
  open: {},
};
export const DEFAULT_READ_ONLY = {
  read_only: {},
};
export const DEFAULT_READ_WRITE = {
  read_write: {},
};
export const DEFAULT_REPLICA_COUNT = {
  replica_count: {
    number_of_replicas: 1,
  },
};
export const DEFAULT_ROLLOVER = {
  rollover: {
    min_doc_count: 5,
  },
};
export const DEFAULT_ROLLUP = {
  // TODO: In future make actual UI
  rollup: {
    jsonString: JSON.stringify(
      {
        ism_rollup: {
          target_index: "rollup-nyc-taxi-data",
          description: "Example rollup job",
          page_size: 200,
          dimensions: [
            {
              date_histogram: {
                source_field: "tpep_pickup_datetime",
                fixed_interval: "1h",
                timezone: "America/Los_Angeles",
              },
            },
            {
              terms: {
                source_field: "PULocationID",
              },
            },
          ],
          metrics: [
            {
              source_field: "passenger_count",
              metrics: [
                {
                  avg: {},
                },
                {
                  sum: {},
                },
                {
                  max: {},
                },
                {
                  min: {},
                },
                {
                  value_count: {},
                },
              ],
            },
          ],
        },
      },
      null,
      4
    ),
  },
};

export const DEFAULT_SHRINK = {
  shrink: {
    percentage_of_source_shards: 0.5,
    force_unsafe: false,
    target_index_name_template: {
      source: "{{ctx.index}}_shrunken",
    },
  },
  force_unsafe_input: "no",
};

export const DEFAULT_SNAPSHOT = {
  snapshot: {
    repository: "example-repository",
    snapshot: "example-snapshot",
  },
};

export const actions = [
  DEFAULT_ALIAS,
  DEFAULT_ALLOCATION,
  DEFAULT_CLOSE,
  DEFAULT_DELETE,
  DEFAULT_FORCE_MERGE,
  DEFAULT_INDEX_PRIORITY,
  DEFAULT_NOTIFICATION,
  DEFAULT_OPEN,
  DEFAULT_READ_ONLY,
  DEFAULT_READ_WRITE,
  DEFAULT_REPLICA_COUNT,
  DEFAULT_ROLLOVER,
  DEFAULT_ROLLUP,
  DEFAULT_SHRINK,
  DEFAULT_SNAPSHOT,
];

export const ISM_TEMPLATE_INPUT_MAX_WIDTH = "400px";
