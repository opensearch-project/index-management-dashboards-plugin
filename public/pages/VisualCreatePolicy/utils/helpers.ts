/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { Action, ISMTemplate, Policy, State, Transition, UIAction } from "../../../../models/interfaces";
import {
  ActionType,
  DEFAULT_ALIAS,
  DEFAULT_ALLOCATION,
  DEFAULT_CLOSE,
  DEFAULT_DELETE,
  DEFAULT_FORCE_MERGE,
  DEFAULT_INDEX_PRIORITY,
  DEFAULT_NOTIFICATION,
  DEFAULT_OPEN,
  DEFAULT_READ_ONLY,
  DEFAULT_READ_WRITE,
  DEFAULT_REPLICA_COUNT,
  DEFAULT_ROLLOVER,
  DEFAULT_ROLLUP,
  DEFAULT_SHRINK,
  DEFAULT_SNAPSHOT,
} from "./constants";
import {
  AliasUIAction,
  AllocationUIAction,
  CloseUIAction,
  DeleteUIAction,
  ForceMergeUIAction,
  IndexPriorityUIAction,
  NotificationUIAction,
  OpenUIAction,
  ReadOnlyUIAction,
  ReadWriteUIAction,
  ReplicaCountUIAction,
  RolloverUIAction,
  RollupUIAction,
  ShrinkUIAction,
  SnapshotUIAction,
} from "../components/UIActions";

export const getConditionContent = (transition: Transition): string => {
  const {
    conditions: {
      min_doc_count: minDocCount = undefined,
      min_index_age: minIndexAge = undefined,
      min_size: minSize = undefined,
      min_rollover_age: minRolloverAge = undefined,
      cron = undefined,
    } = {},
  } = transition;
  if (minSize != undefined) return `Minimum index size is ${minSize}`;
  if (minDocCount != undefined) return `Minimum index doc count is ${minDocCount}`;
  if (minIndexAge != undefined) return `Minimum index age is ${minIndexAge}`;
  if (minRolloverAge != undefined) return `Minimum rollover age is ${minRolloverAge}`;
  if (cron != undefined) return `After cron expression "${cron.cron.expression}" in ${cron.cron.timezone}`;
  return "No condition";
};

export const getUIActionFromData = (action: Action): UIAction<any> => {
  const actionType = Object.keys(action)
    .filter((key) => key !== "timeout" && key !== "retry")
    .pop();
  if (!actionType) throw new Error(`Failed to get action using type [${actionType}]`);
  const uiAction = getUIAction(actionType);
  return uiAction.clone(action);
};

export const getUIAction = (actionType: string): UIAction<any> => {
  switch (actionType) {
    case ActionType.Alias:
      return new AliasUIAction(DEFAULT_ALIAS);
    case ActionType.Allocation:
      return new AllocationUIAction(DEFAULT_ALLOCATION);
    case ActionType.Close:
      return new CloseUIAction(DEFAULT_CLOSE);
    case ActionType.Delete:
      return new DeleteUIAction(DEFAULT_DELETE);
    case ActionType.ForceMerge:
      return new ForceMergeUIAction(DEFAULT_FORCE_MERGE);
    case ActionType.IndexPriority:
      return new IndexPriorityUIAction(DEFAULT_INDEX_PRIORITY);
    case ActionType.Notification:
      return new NotificationUIAction(DEFAULT_NOTIFICATION);
    case ActionType.Open:
      return new OpenUIAction(DEFAULT_OPEN);
    case ActionType.ReadOnly:
      return new ReadOnlyUIAction(DEFAULT_READ_ONLY);
    case ActionType.ReadWrite:
      return new ReadWriteUIAction(DEFAULT_READ_WRITE);
    case ActionType.ReplicaCount:
      return new ReplicaCountUIAction(DEFAULT_REPLICA_COUNT);
    case ActionType.Rollover:
      return new RolloverUIAction(DEFAULT_ROLLOVER);
    case ActionType.Rollup:
      return new RollupUIAction(DEFAULT_ROLLUP);
    case ActionType.Shrink:
      return new ShrinkUIAction(DEFAULT_SHRINK);
    case ActionType.Snapshot:
      return new SnapshotUIAction(DEFAULT_SNAPSHOT);
    default:
      throw new Error(`Action type [${actionType}] not supported`);
  }
};

export const getActionOptions = (actionRepoSingleton: ActionRepository) => {
  return actionRepoSingleton.getAllActionTypes().map((key) => {
    let uiAction;
    let text = key
      .split("_")
      .map((str) => capitalizeFirstLetter(str))
      .join(" ");
    try {
      uiAction = getUIAction(key)?.customDisplayText;
      if (uiAction !== undefined) text = uiAction;
    } catch (e) {}
    return {
      value: key,
      text: text,
    };
  });
};

class ActionRepository {
  repository: { [actionType: string]: [new (action: Action) => UIAction<any>, Action] } = {
    alias: [AliasUIAction, DEFAULT_ALIAS],
    allocation: [AllocationUIAction, DEFAULT_ALLOCATION],
    close: [CloseUIAction, DEFAULT_CLOSE],
    delete: [DeleteUIAction, DEFAULT_DELETE],
    force_merge: [ForceMergeUIAction, DEFAULT_FORCE_MERGE],
    index_priority: [IndexPriorityUIAction, DEFAULT_INDEX_PRIORITY],
    notification: [NotificationUIAction, DEFAULT_NOTIFICATION],
    open: [OpenUIAction, DEFAULT_OPEN],
    read_only: [ReadOnlyUIAction, DEFAULT_READ_ONLY],
    read_write: [ReadWriteUIAction, DEFAULT_READ_WRITE],
    replica_count: [ReplicaCountUIAction, DEFAULT_REPLICA_COUNT],
    rollover: [RolloverUIAction, DEFAULT_ROLLOVER],
    rollup: [RollupUIAction, DEFAULT_ROLLUP],
    shrink: [ShrinkUIAction, DEFAULT_SHRINK],
    snapshot: [SnapshotUIAction, DEFAULT_SNAPSHOT],
  };

  getAllActionTypes = () => {
    return Object.keys(this.repository);
  };

  registerAction = (actionType: string, uiActionCtor: new (action: Action) => UIAction<any>, defaultAction: Action) => {
    if (this.repository.hasOwnProperty(actionType)) {
      throw new Error(`Cannot register an action twice in the repository [type=${actionType}]`);
    }

    this.repository[actionType] = [uiActionCtor, defaultAction];
  };

  getUIActionFromData = (action: Action) => {
    const actionType = Object.keys(action)
      .filter((key) => key !== "timeout" && key !== "retry")
      .pop();
    if (!actionType) throw new Error(`Failed to get action using type [${actionType}]`);
    const uiAction = this.getUIAction(actionType);
    return uiAction.clone(action);
  };

  getUIAction = (actionType: string): UIAction<any> => {
    const uiAction = this.repository[actionType];
    if (!uiAction) throw new Error(`Action type [${actionType}] not supported`);
    return new uiAction[0](uiAction[1]);
  };
}

export const actionRepoSingleton = new ActionRepository();

// Takes in the ismTemplates which could be a single object, array, or not defined and returns them reformatted as a list
export const convertTemplatesToArray = (ismTemplates: ISMTemplate[] | ISMTemplate | null | undefined): ISMTemplate[] => {
  const templates = [];
  // policy.ism_template can be an array of templates or a single template as an object or null
  if (Array.isArray(ismTemplates)) {
    templates.push(...ismTemplates);
  } else if (ismTemplates) {
    templates.push(ismTemplates);
  }
  return templates;
};

export const capitalizeFirstLetter = ([first, ...rest]: string, locale = navigator.language) =>
  first.toLocaleUpperCase(locale) + rest.join("");

export const getOrderInfo = (
  state: State | null,
  states: State[]
): { order: string; afterBeforeState: string; disableOrderSelections: boolean } => {
  const isEditing = !!state;
  const editingStateIdx = states.findIndex((s) => s === state);
  let afterBeforeState = "";
  let order = "after";
  let disableOrderSelections = false;
  if (isEditing) {
    // If there is only one existing state then we are editing the only state and there is nothing to order after/before
    if (states.length === 1) {
      disableOrderSelections = true;
    } else {
      // If the editing state is the first state then we are "Add before $secondState"
      if (editingStateIdx === 0) {
        afterBeforeState = states[editingStateIdx + 1]?.name;
        order = "before";
      } else {
        // otherwise we are "Add after $previousState"
        afterBeforeState = states[editingStateIdx - 1]?.name;
      }
    }
  } else {
    // when creating a new state
    // If there are no existing states then there is nothing to order after/before
    if (!states.length) {
      disableOrderSelections = true;
    } else {
      // Else always order it after the last state
      afterBeforeState = states[states.length - 1]?.name;
    }
  }

  return { order, afterBeforeState, disableOrderSelections };
};

export const getUpdatedStates = (
  state: State,
  editingState: State | null,
  states: State[],
  order: string,
  afterBeforeState: string
): State[] => {
  const someStates: State[] = [];
  const isEditing = !!editingState;
  if ((isEditing && states.length === 1) || (!isEditing && !states.length)) return [state];

  states.forEach((s, idx) => {
    if (s.name === afterBeforeState && order === "before") someStates.push(state);
    if (s.name !== editingState?.name) someStates.push(s);
    if (s.name === afterBeforeState && order === "after") someStates.push(state);
  });

  return someStates;
};

export const getUpdatedPolicy = (
  currentPolicy: Policy,
  updatedState: State,
  editingState: State | null,
  currentStates: State[],
  order: string,
  afterBeforeState: string
): Policy => {
  const updatedStates = getUpdatedStates(updatedState, editingState, currentStates, order, afterBeforeState);
  let defaultState = currentPolicy.default_state;
  // If there is 1 state, change it to this state
  if (updatedStates.length === 1) {
    defaultState = updatedStates[0].name;
  }
  // Change the default state if the state being edited was the default state
  if (editingState && editingState.name === defaultState) {
    // don't bother checking if the name itself changed, just set it regardless to the value from the new state
    defaultState = updatedState.name;
  }

  return {
    ...currentPolicy,
    states: updatedStates,
    default_state: defaultState,
  };
};
