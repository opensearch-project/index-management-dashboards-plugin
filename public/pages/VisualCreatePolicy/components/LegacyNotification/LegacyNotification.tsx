/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { EuiFormRow, EuiCodeEditor } from "@elastic/eui";
import "brace/theme/github";
import "brace/mode/json";
import { DarkModeConsumer } from "../../../../components/DarkMode";
import { DEFAULT_LEGACY_ERROR_NOTIFICATION } from "../../utils/constants";

interface LegacyNotificationProps {
  notificationJsonString: string;
  onChangeNotificationJsonString: (str: string) => void;
  actionNotification?: boolean;
  isInvalid?: boolean;
}

const LegacyNotification = ({
  notificationJsonString,
  onChangeNotificationJsonString,
  actionNotification = false,
  isInvalid = false, // TODO: default to false for error notification for now, but add validation logic for it
}: LegacyNotificationProps) => {
  return (
    <>
      <EuiFormRow fullWidth isInvalid={isInvalid} error={null} style={{ maxWidth: "100%" }}>
        <DarkModeConsumer>
          {(isDarkMode) => (
            <EuiCodeEditor
              mode="json"
              placeholder={JSON.stringify(DEFAULT_LEGACY_ERROR_NOTIFICATION, null, 4)}
              theme={isDarkMode ? "sense-dark" : "github"}
              width="100%"
              value={notificationJsonString}
              onChange={onChangeNotificationJsonString}
              setOptions={{ fontSize: "14px" }}
              aria-label="Code Editor"
              height="300px"
              data-test-subj={actionNotification ? "create-policy-action-legacy-notification" : "create-policy-legacy-notification"}
            />
          )}
        </DarkModeConsumer>
      </EuiFormRow>
    </>
  );
};

export default LegacyNotification;
