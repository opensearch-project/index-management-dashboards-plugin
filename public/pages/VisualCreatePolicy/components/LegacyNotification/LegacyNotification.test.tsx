/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import LegacyNotification from "./LegacyNotification";
import { DEFAULT_LEGACY_ERROR_NOTIFICATION } from "../../utils/constants";

describe("<LegacyNotification /> spec", () => {
  it("renders the component", () => {
    const { container } = render(
      <LegacyNotification
        notificationJsonString={JSON.stringify(DEFAULT_LEGACY_ERROR_NOTIFICATION)}
        onChangeNotificationJsonString={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
