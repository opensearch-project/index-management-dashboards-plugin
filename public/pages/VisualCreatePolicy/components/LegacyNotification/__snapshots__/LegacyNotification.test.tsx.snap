// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegacyNotification /> spec renders the component 1`] = `
<div
  class="euiFormRow euiFormRow--fullWidth"
  id="some_html_id-row"
  style="max-width: 100%;"
>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      class="euiCodeEditorWrapper"
      data-test-subj="create-policy-legacy-notification"
      style="width: 100%; height: 300px;"
    >
      <button
        class="euiCodeEditorKeyboardHint"
        data-test-subj="codeEditorHint"
        id="some_html_id"
      >
        <p
          class="euiText"
        >
          Press Enter to start editing.
        </p>
        <p
          class="euiText"
        >
          When you're done, press Escape to stop editing.
        </p>
      </button>
      <div
        class=" ace_editor ace-tm"
        id="some_html_id"
        style="width: 100%; height: 300px; font-size: 14px;"
      >
        <textarea
          aria-label="Code Editor"
          autocapitalize="off"
          autocorrect="off"
          class="ace_text-input"
          spellcheck="false"
          style="opacity: 0;"
          tabindex="-1"
          wrap="off"
        />
        <div
          aria-hidden="true"
          class="ace_gutter"
        >
          <div
            class="ace_layer ace_gutter-layer ace_folding-enabled"
          />
          <div
            class="ace_gutter-active-line"
          />
        </div>
        <div
          class="ace_scroller"
        >
          <div
            class="ace_content"
          >
            <div
              class="ace_layer ace_print-margin-layer"
            >
              <div
                class="ace_print-margin"
                style="left: 4px; visibility: visible;"
              />
            </div>
            <div
              class="ace_layer ace_marker-layer"
            />
            <div
              class="ace_layer ace_text-layer"
              style="padding: 0px 4px;"
            />
            <div
              class="ace_layer ace_marker-layer"
            />
            <div
              class="ace_layer ace_cursor-layer ace_hidden-cursors"
            >
              <div
                class="ace_cursor"
              />
            </div>
          </div>
        </div>
        <div
          class="ace_scrollbar ace_scrollbar-v"
          style="display: none; width: 20px;"
        >
          <div
            class="ace_scrollbar-inner"
            style="width: 20px;"
          />
        </div>
        <div
          class="ace_scrollbar ace_scrollbar-h"
          style="display: none; height: 20px;"
        >
          <div
            class="ace_scrollbar-inner"
            style="height: 20px;"
          />
        </div>
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
        >
          <div
            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
          />
          <div
            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
          >
            XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
